sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -167,10 +169,43 @@ class MapboxMap internal constructor(
     onStyleLoaded: Style.OnStyleLoaded? = null
   ) {
     this.style = style
-    styleExtension.images.forEach {
-      it.bindTo(style)
-    }
+    var geojsonNotParsedCount = 0
+    var hasGeojsonSources = false
     styleExtension.sources.forEach {
+      if (it is GeoJsonSource) {
+        hasGeojsonSources = true
+        if (it.geoJsonParsed) {
+          it.bindTo(style)
+        } else {
+          geojsonNotParsedCount++
+          it.geoJsonParsedListener = {
+            geojsonNotParsedCount--
+            it.bindTo(style)
+            if (geojsonNotParsedCount == 0) {
+              loadAllExceptGeojsonSources(style, styleExtension, onStyleLoaded)
+            }
+          }
+        }
+      } else {
+        it.bindTo(style)
+      }
+    }
+    if (hasGeojsonSources) {
```

Comment:
We can merge the else block here with `if(!hasGeojsonSources || geojsonNotParsedCount == 0)`

Reply:
Revisited overall logic here.